samples/99_zil_interpreter/app/parsers/scanning_parser_v1.rb

```diff
@@ -0,0 +1,145 @@
+class ScanningParser
+
+  def initialize(work_dir)
+    @work_dir = work_dir
+  end
+
+  def parse(args, file_name)
+    args.state.ztokens ||= {}
+    file = args.gtk.read_file(@work_dir + file_name)
+
+    lines = []
+    for line in file.each_line
+      lines << line
+    end
+
+    log "Scanning '" + file_name + "' ..."
+    scanner = Scanner.new(file_name, lines)
+
+    i = 0
+    while scanner.scan_to('<')
+      block = scanner.read_block
+
+      # block should be < to > statement, but some multiline tokens starting with ( to ) break the logic
+
+      #if block.include?('PALANTIR') == true # grep out PALANTIR objects to show to kfischer
+      #  log "----- " + file_name + " STATEMENT"+ i.to_s + " -----"
+      #  log block
+      #end
+
+      i += 1
+    end
+
+    log "Scan completed."
+
+    #i = 0
+    #num_lines = lines.length
+    #while i < num_lines
+    #  i += 1
+    #end
+  end
+end
+
+class Scanner
+  def initialize(file_name, lines)
+    @lines = lines
+    @file_name = file_name
+    reset
+  end
+
+  def reset
+    @line_num = 0
+    @line_index = -1
+  end
+
+  def scan_to(scan_char)
+    found = false
+
+    num_lines = @lines.length
+    while @line_num < num_lines
+
+      line = @lines[@line_num]
+      #log "[SCAN:#{@file_name}:#{@line_num}.#{line.length}] " + line
+
+      while @line_index < line.length - 1
+        @line_index += 1
+        line_char = line[@line_index]
+
+        if line_char == scan_char
+          #log "Found '" + scan_char + "' at " + @line_num.to_s + ":" + @line_index.to_s
+          found = true
+          break
+        end
+
+      end
+
+      if found
+        #log "Found '" + cur_char
+        break
+      else
+        @line_index = -1
+      end
+
+      @line_num += 1
+    end
+
+    found
+  end
+
+  def read_block

```

Comment:
the  `Scanner` class should not  (in my opinion) handle blocks, that should be left to the parser

Reply:
Agreed.  Fixed in update.